tests/integration-service/namespace-backed-environments.go

```diff
@@ -260,6 +262,108 @@ var _ = framework.IntegrationServiceSuiteDescribe("Namespace-backed Environment
 				Expect(err.Error()).To(ContainSubstring(constants.EphemeralEnvAbsenceErrorString))
 			})
 		})
+
+		It("creates a new IntegrationTestScenario with ephemeral environment", FlakeAttempts(3), func() {
+			var err error
+			newIntegrationTestScenario, err = f.AsKubeAdmin.IntegrationController.CreateIntegrationTestScenarioWithEnvironment(applicationName, testNamespace, gitURL, revisionForNBE, pathInRepoForNBE, userPickedEnvironment)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("updates the Snapshot with the re-run label for the new scenario", FlakeAttempts(3), func() {
+			updatedSnapshot := snapshot.DeepCopy()
+			err := metadata.AddLabels(updatedSnapshot, map[string]string{snapshotRerunLabel: newIntegrationTestScenario.Name})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(f.AsKubeAdmin.IntegrationController.PatchSnapshot(snapshot, updatedSnapshot)).Should(Succeed())
+			Expect(metadata.GetLabelsWithPrefix(updatedSnapshot, snapshotRerunLabel)).NotTo(BeEmpty())
+		})
+
+		When("An snapshot is updated with a rerun label for a given scenario", func() {
+			It("checks if the rerun label was removed from the Snapshot", FlakeAttempts(3), func() {
+				Eventually(func() error {
+					snapshot, err = f.AsKubeAdmin.IntegrationController.GetSnapshot(snapshot.Name, "", "", testNamespace)
+					if err != nil {
+						return fmt.Errorf("encountered errror while getting Snapshot %s/%s", snapshot.Name, snapshot.Namespace)
```

Comment:
possibly could return `err` here as well?

Reply:
Ack, added the `err` here.